From 1d0387103dfeecdcd01d35c9fe61b6d0ff712bba Mon Sep 17 00:00:00 2001
From: Omri Itach <omrii@marvell.com>
Date: Wed, 7 Aug 2013 17:02:21 +0300
Subject: [PATCH 0941/1825] alp: added routine to extract sscg status from
 sample at reset

https://github.com/MISL-EBU-System-SW/misl-windriver.git linux-3.4.69-14t2-read
commit f89f11d17c3e6d399234cada00e2ed766d2a3702

	- Added routine mvCtrlIsSscgEnabled() to return sscg status from S@R

Change-Id: I58ea429bf925b90bbbeb2113eb6d951ae2cc3828
Signed-off-by: Omri Itach <omrii@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/2970
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Kosta Zertsekel <konszert@marvell.com>
Signed-off-by: Zhong Hongbo <hongbo.zhong@windriver.com>
---
 .../avanta_lp_family/ctrlEnv/mvCtrlEnvLib.c        |   11 +++++++++++
 .../avanta_lp_family/ctrlEnv/mvCtrlEnvLib.h        |    1 +
 2 files changed, 12 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/mvCtrlEnvLib.c b/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/mvCtrlEnvLib.c
index 22dc409..031ed85 100755
--- a/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/mvCtrlEnvLib.c
+++ b/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/mvCtrlEnvLib.c
@@ -137,6 +137,17 @@ MV_U32 mvCtrlGetCpuNum(MV_VOID)
 		return cpu1Enabled;
 }
 
+MV_BOOL mvCtrlIsSscgEnabled(MV_VOID)
+{
+	MV_U32 sscgDisabled;
+
+	if (mvCtrlSatRRead(MV_SATR_SSCG_DISABLE, &sscgDisabled) != MV_OK) {
+		mvOsPrintf("%s: Error: failed to read SSCG status\n", __func__);
+		return MV_TRUE; /* if error reading S@R, assume sscg is default (enabled) */
+	} else
+		return sscgDisabled == 0x1 ? MV_FALSE : MV_TRUE;
+}
+
 /*******************************************************************************
 * mvCtrlIsValidSatR
 *
diff --git a/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/mvCtrlEnvLib.h b/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/mvCtrlEnvLib.h
index 5a83e19..e7d9126 100755
--- a/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/mvCtrlEnvLib.h
+++ b/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/mvCtrlEnvLib.h
@@ -277,6 +277,7 @@ MV_VOID mvCtrlSatrInit(MV_VOID);
 MV_VOID mvCtrlSysConfigInit(MV_VOID);
 MV_U32 mvCtrlSysConfigGet(MV_CONFIG_TYPE_ID configField);
 MV_U32 mvCtrlGetCpuNum(MV_VOID);
+MV_BOOL mvCtrlIsSscgEnabled(MV_VOID);
 MV_U32 mvCtrlGetQuadNum(MV_VOID);
 MV_STATUS mvCtrlUpdatePexId(MV_VOID);
 MV_BOOL mvCtrlIsValidSatR(MV_VOID);
-- 
1.7.5.4

